cpp/oneapi/dal/graph/service_functions_test.cpp

```diff
@@ -0,0 +1,166 @@
+#include "oneapi/dal/test/engine/common.hpp"
+
+#include "oneapi/dal/graph/service_functions.hpp"
+#include "oneapi/dal/graph/undirected_adjacency_vector_graph.hpp"
+#include "oneapi/dal/io/graph_csv_data_source.hpp"
+#include "oneapi/dal/io/load_graph.hpp"
+
+namespace oneapi::dal::graph::test {
+
+namespace dal = oneapi::dal;
+
+using my_graph_type = dal::preview::undirected_adjacency_vector_graph<>;
+const dal::preview::load_graph::descriptor<dal::preview::edge_list<int32_t>, my_graph_type>
+    desc;
+
+auto get_data_source (const std::string& name) {
+	const std::string name_path = "cpp/oneapi/dal/graph/test_data/" + name;
+	const dal::preview::graph_csv_data_source ds(name_path);
+	return ds;
+}
+
+auto check_neighbors (const std::string& name, 
+	const dal::preview::vertex_size_type<my_graph_type> *correct_neigh,
+	const dal::preview::vertex_size_type<my_graph_type> vertices_num) {
+	int correct_neigh_num = 0;
+	dal::preview::vertex_size_type<my_graph_type> neigh_index = 0;
+	const auto my_graph = dal::preview::load_graph::load(desc, get_data_source(name));
+	for (dal::preview::vertex_size_type<my_graph_type> j = 0; j < vertices_num; j++) {
+		const auto neigh = dal::preview::get_vertex_neighbors(my_graph, j);
+		for (auto i = neigh.first; i != neigh.second; ++i) {
+            if (*i == correct_neigh[neigh_index]) correct_neigh_num++;	
+        	neigh_index++;
+        }
+    }
+    return correct_neigh_num;
+}
+
+TEST_CASE("throws if input file is empty") {
+	REQUIRE_THROWS_WITH(dal::preview::load_graph::load(desc, get_data_source("graph0.csv")),"Empty edge list");
+}
+
+TEST_CASE("throws if input file does not exist") {
+	REQUIRE_THROWS_WITH(dal::preview::load_graph::load(desc, get_data_source("nonexistgraph.csv")),"File not found");
+}
+////////////////////////////////
+
+TEST_CASE("get_vertex_count for the graph with 2 vertices") {
+	const auto my_graph = dal::preview::load_graph::load(desc, get_data_source("graph1.csv"));
+	REQUIRE(dal::preview::get_vertex_count(my_graph) == 2);
```

Comment:
This test case does not test `service_functions`, but tests `load_graph`. It checks whether function `load` correctly interprets file like that:
```csv
0 1 
```

Right approach: 
- Create graph from in-memory data with specific vertex count and check if `get_vertext_count` returns expected value.
- Create separate test in `io/load_graph` module to cover `load` function